dash/include/dash/Meta.h

```diff
@@ -90,6 +90,19 @@ struct is_container_compatible :
          >
 { };
 
+/**
+ * Type trait indicating whether a type can be used for global atomic
+ * operations.
+ */
+template <typename T>
+struct is_atomic_compatible
+: public std::integral_constant<
+           bool,
+              dash::is_container_compatible<T>::value
+           && sizeof(T) <= sizeof(std::size_t)
+         >
```

Comment:
Right, that's what I meant with `dash::is_global_atomic_compatible` to distinguish between "atomic operations supported by STL" and "atomic operations supported by DART".
Because some types are fine for `std::atomic<T>` but not supported by DART reduce operations, like `struct DoublePair { double; double; }`.
It would be okay to use `dash::Atomic<DoublePair>`, though:
Compare-exchange would then be supported for local memory (probably lock-based) but not across units.
Still, I stand by my point: With the current implementation of DART and dash::Atomic, the type trait `is_atomic_compatible` in this PR is correct.